### PR TITLE
WIP: Single shell/noreg redux

### PR DESCRIPTION
**Pull request by arokem**
I need the single shell-free water DTI method for my own work, so I am trying to revive this old unmerged work from @mvgolub, with the hopes that I can push it past the finish line. This is not exactly the same code as in #2087, but from [this comment](https://github.com/dipy/dipy/pull/2087#issuecomment-859688187), I gather that it is the most up-to-date code to start working from. I believe that the main thing that I will need to do here is to write tests for all the new code. For now, marked as WIP, until I can sort that out.

### CODE REVIEW

**Comment by mvgolub** [comment]
> I need the single shell-free water DTI method for my own work, so I am trying to revive this old unmerged work from @mvgolub, with the hopes that I can push it past the finish line. This is not exactly the same code as in #2087, but from [this comment](https://github.com/dipy/dipy/pull/2087#issuecomment-859688187), I gather that it is the most up-to-date code to start working from. I believe that the main thing that I will need to do here is to write tests for all the new code. For now, marked as WIP, until I can sort that out.

Hi @arokem , based on the comment you mentioned:

> EDIT: the function fernet_iter was too long and confusing, I decided to separate it into three functions instead: fwmd_iter, fws0_iter and fwhy_iter, each corresponding to a specific free water initialization method.

I realized that I did these changes on my local machine but I forgot to commit them. And in the current code, the long `fernet_iter` function is still present instead. Have you already started writing tests for this function? If yes, I will revert the changes in my local code and keep working from that, if not, I can still commit my recent changes. It depends on what you think is best: keep the `fernet_iter` function as it is, or separate it if you think it is long and unreadable.

File: dipy/reconst/fwdti.py [hunk 1]
```diff
@@ -800,6 +819,253 @@ def cholesky_to_lower_triangular(R):
...
+        fmin = np.clip(fmin, 0, 1)
+        fmax = np.clip(fmax, 0, 1)
+
+        # Estimate for tissue water fraction based on S0
+        fs0 = 1 - np.log(S0 / St) / np.log(Sw / St)
+        fs0_clip = np.clip(fs0, fmin, fmax)
+
+        if method == 's0':
+            # DTI applied to tissue signal contribution
+            fw = 1 - fs0
```
Comment (mvgolub) [inline dipy/reconst/fwdti.py:985]: I think this is meant to be `fw = 1 - fs0_clip`, have to change it later

**Comment by arokem** [comment]
Hi @mvgolub : if you could commit your changes that would be great!

**Comment by mvgolub** [comment]
Hi, I already pushed the most recent changes into my remote dipy fork.

**Comment by arokem** [comment]
Hi @mvgolub : what branch on your fork should I look at for your most recent changes? Is it in the default branch on your fork (the one called "placeholder")?

**Comment by mvgolub** [comment]
Hi @arokem , I pushed the most recent changes to the  `singleshell-fwdti-noreg`  branch of my dipy fork, here https://github.com/mvgolub/dipy/tree/singleshell-fwdti-noreg

**Comment by arokem** [comment]
Great. Thanks! I'll pull from there.

**Comment by arokem** [comment]
Closed in favor of #2705, because of the complexity of the rebase here.

**Pull request closed without merging**